src/blockrelay/graphene.cpp

```diff
@@ -46,9 +46,12 @@ CGrapheneBlock::CGrapheneBlock(const CBlockRef pblock,
         grapheneSetVersion = 1;
     }
 
-    useSipHash = _useSipHash;
-    if (useSipHash)
+    version = _version;
+    if (version >= 2)
+    {
         FillShortTxIDSelector();
+        grapheneSetVersion = 1;
+    }
 
```

Comment:
this is a little untitutive to me..there is a version , then a graphenesetversion then further down there is grapheneVersion?  are they all different or is version also == grapheneVersion?  if version means grapheneVersion then maybe call it that...would make the code more readable IMHO.

Reply:
I agree that it's cumbersome, but let me explain the reasoning a little; then perhaps you can suggest a better solution? There are now three objects related to graphene that have a (separate) version number: `CGrapheneBlock`, `CGrapheneSet`, and `CIblt`. I thought about just using the same version number for each, but it ended up not making sense to me because then you would need to, for example, bump the version of `CIblt` every time you make a change to `CGrapheneBlock`. So that's my reasoning for having two different variables: `grapheneVersion` and `grapheneSetVersion`, in general they are different. Regarding the generic name `version`, each one of those objects uses `version` as a member variable where it is unambiguous.  I use a more descriptive variable name in cases where the class associated with the version is not necessarily clear. Does that makes sense?

Reply:
on a related note, if grapheneSetVersion is changing from a prior recent PR to this one, we can combine that into the same version number, since we did not make a release or merge to our release branch.

Reply:
I would rather not do that only because version number checks are performed in a few different places, leaving room to introduce an error if changed. Also, downstream patches based on this one bump the version number assuming the current value.